### PR TITLE
feat: improve transactions signing UX for mobile

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1387

- display a modal to sign transactions while using the app on desktop with a mobile connection (QRCode)
- automatically open Keplr app to sign transactions on mobile
- create a performant context to share global variables

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://aggressive-rose.surge.sh/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/components/organisms/RegistryLayout.tsx [hunk 1]
```diff
@@ -97,9 +96,7 @@ const RegistryLayout: React.FC<React.PropsWithChildren<unknown>> = () => {
           <Box display="flex" justifyContent="center" alignItems="center">
             {chainId && loaded && wallet?.address && disconnect && (
               <UserMenuItem
-                address={
-                  isMobile ? truncate(wallet?.address) : wallet?.shortAddress
-                }
+                address={truncate(wallet?.address)}
```
Comment (flagrede) [inline web-registry/src/components/organisms/RegistryLayout.tsx:99]: A small request from @erikalogie to have a consistent address truncating between desktop and mobile.

File: web-registry/src/components/organisms/WalletButton/WalletButton.tsx [hunk 2]
```diff
@@ -19,8 +22,12 @@ const WalletButton: React.FC = () => {
   const styles = useWalletButtonStyles();
   const { wallet, connect, loaded, error, walletConnectUri } = useWallet();
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isWaitingForSigning, setGlobalStore] = useGlobalStore(
+    store => store['isWaitingForSigning'],
```
Comment (flagrede) [inline web-registry/src/components/organisms/WalletButton/WalletButton.tsx:26]: A new global context to share values between deeply nested components.

**Comment by flagrede** [comment]
@erikalogie you can try signing improvements with either Desktop + QRCode or directly on mobile here: https://aggressive-rose.surge.sh/

**Review by mhagel — APPROVED** [review]
I was able to test desktop w/ wallet connect, but the test site did not load on my phone.

**Review by blushi — APPROVED** [review]
LGTM
Could test this successfully on both desktop and mobile

**Comment by erikalogie** [comment]
I added our shared testing account on my keplr mobile app and tested this by connecting the keplr mobile app to the desktop site. When trying to purchase ecocredits I got this error: https://monosnap.com/file/3AUnHkDF0IMZMShWoC6peOFQXxig9P Is this expected?

Also on mobile I should not be able to sign the transaction as test nets are not available on the keplr app, correct?

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach master

**Pull request merged**